CMakeLists.txt

```diff
@@ -148,19 +153,28 @@ if(NOT HAVE_SYS_UIO_H_01)
   set(HAVE_SYS_UIO_H_01 0)
 endif(NOT HAVE_SYS_UIO_H_01)
 
-if (SNAPPY_FUZZING_BUILD)
-  if (NOT "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+if(SNAPPY_FUZZING_BUILD OR SNAPPY_OSSFUZZ_BUILD)
+  if(NOT "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
     message(WARNING "Fuzzing builds are only supported with Clang")
-  endif (NOT "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+  endif(NOT "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
 
-  if(NOT CMAKE_CXX_FLAGS MATCHES "-fsanitize=address")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address")
-  endif(NOT CMAKE_CXX_FLAGS MATCHES "-fsanitize=address")
 
   if(NOT CMAKE_CXX_FLAGS MATCHES "-fsanitize=fuzzer-no-link")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=fuzzer-no-link")
   endif(NOT CMAKE_CXX_FLAGS MATCHES "-fsanitize=fuzzer-no-link")
-endif (SNAPPY_FUZZING_BUILD)
+endif(SNAPPY_FUZZING_BUILD OR SNAPPY_OSSFUZZ_BUILD)
+
+if(SNAPPY_OSSFUZZ_BUILD)
+  if(NOT DEFINED ENV{LIB_FUZZING_ENGINE})
+    message(SEND_ERROR "OSS-Fuzz builds require LIB_FUZZING_ENGINE to be set")
+  endif(NOT DEFINED ENV{LIB_FUZZING_ENGINE})
+endif(SNAPPY_OSSFUZZ_BUILD)
```

Comment:
Error if the ossfuzz build has not defined `LIB_FUZZING_ENGINE` environment variable.